src/config.rs

```diff
@@ -140,22 +181,23 @@ impl Config {
     }
 }
 
-fn read_toml(filename: &str) -> toml::Value {
+fn read_toml(filename: &str) -> Result<toml::Value, ConfigError> {
     let mut input = String::new();
-    File::open(filename).and_then(|mut f| {
+    match File::open(filename).and_then(|mut f| {
         f.read_to_string(&mut input)
-    }).unwrap();
+    }) {
+        Ok(_) => {}
+        Err(e) => {
+            return Err(ConfigError::Io(e, filename.to_owned()));
+        }
+    }
+
     let mut parser = toml::Parser::new(&input);
     match parser.parse() {
-        Some(toml) => toml::Value::Table(toml),
+        Some(toml) => Ok(toml::Value::Table(toml)),
         None => {
-            for err in &parser.errors {
-                let (loline, locol) = parser.to_linecol(err.lo);
-                let (hiline, hicol) = parser.to_linecol(err.hi);
-                println!("{}:{}:{}-{}:{} error: {}",
-                         filename, loline, locol, hiline, hicol, err.desc);
-            }
-            panic!("Errors in config")
+            let tmp = parser.errors.get(0).unwrap();
+            Err(ConfigError::TomlError(toml::ParserError{lo:0,hi:0,desc:tmp.desc.clone()}, filename.to_owned()))
```

Comment:
I'd rather `remove` the element but retain all its information (and possibly file a bug about this struct not deriving `Clone`). And rereading the docs maybe the element we want is the last one not the first. The toml-rs library is just weird.